integration/tests/epochs/suite.go

```diff
@@ -96,3 +101,246 @@ func (s *Suite) TearDownTest() {
 		s.cancel()
 	}
 }
+
+// StakeNode will generate initial keys needed for a SN/LN node and onboard this node using the following steps;
+// 1. Generate keys (networking, staking, machine)
+// 2. Create a new lease account, this will be the staking account
+// 3. Transfer token amount for the given role to the staking account
+// 4. Add additional funds to staking account for storage
+// 5. Create Staking collection for node
+// 6. Register node using staking collection object
+func (s *Suite) StakeNode(role flow.Role) {
+	stakingAccountKey, networkingKey, stakingKey, _, machineAccountPubKey := s.generateAccountKeys()
+	nodeID := flow.MakeID(stakingKey.PublicKey().Encode())
+	fullAccountKey := sdk.NewAccountKey().
+		SetPublicKey(stakingAccountKey.PublicKey()).
+		SetHashAlgo(sdkcrypto.SHA2_256).
+		SetWeight(sdk.AccountKeyWeightThreshold)
+
+	// create staking account
+	stakingAccountAddress, err := s.createNewLeaseAccount(fullAccountKey)
+	require.NoError(s.T(), err)
+
+	// fund account with token amount to stake
+	result, err := s.transferLeaseTokens(role, stakingAccountAddress)
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	// fund account for storage
+	result, err = s.fundAccount(stakingAccountAddress, "10.0")
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	stakingAccount, err := s.client.GetAccount(stakingAccountAddress)
+	require.NoError(s.T(), err)
+
+	// create staking collection
+	_, err = s.createStakingCollection(stakingAccountKey, stakingAccount)
+	require.NoError(s.T(), err)
+	require.NoError(s.T(), result.Error)
+
+	tokenAmount, err := s.client.TokenAmountByRole(flow.RoleConsensus.String())
+	require.NoError(s.T(), err)
+
+	encMachinePubKey, err := flow.EncodeRuntimeAccountPublicKey(machineAccountPubKey)
```

Comment:
This is only needed for SN/LN nodes, but we want to be able to stake any node role. 

I think this is OK, because the [transaction](https://github.com/onflow/flow-core-contracts/blob/master/transactions/stakingCollection/register_node.cdc#L20) should handle it. But we could add a comment that this will be ignored for nodes that don't need a machine account. When we return this data to the caller, we should avoid passing a value for machine account key for node roles which don't use a machine account.

Reply:
https://github.com/onflow/flow-go/pull/1320/commits/4b1a90ffb551beb5f4f9cbb833424a62bf9e10b0